### PR TITLE
#128 Added ability of drawing nonogram with numbers around

### PR DESCRIPTION
**Pull request by NikitaSukhanov**

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        }
+    }
+
+    /**
+     * Function for drawing left field of nonogram.
+     *
+     * @param canvas   - canvas received in 'onDraw'
+     * @param nonogram - nonogram to draw
+     */
+    private static void drawLeftField(Canvas canvas, final INonogramExtended nonogram) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java]: The length of this line is over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+    }
+
+    /**
+     * Function for drawing top field of nonogram.
+     *
+     * @param canvas   - canvas received in 'onDraw'
+     * @param nonogram - nonogram to draw
+     */
+    private static void drawTopField(Canvas canvas, final INonogramExtended nonogram) {
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java]: The length of this line is over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        final float startY = nonogram.getMainFieldTopLeftY();
+
+        String text;
+        int length, index;
+        int textWidth, textHeight;
+
+        _cellsPainter.setColor(Color.WHITE);
+        _backgroundRect.top = startY - nonogram.getTopFieldHeight();
+        _backgroundRect.left = startX;
+        _backgroundRect.bottom = _backgroundRect.top + nonogram.getTopFieldHeight();
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java]: The length of this line is over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        _backgroundRect.right = _backgroundRect.left + nonogram.getTopFieldWidth();
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java]: The length of this line is over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+    /**
+     * Set parameters for painters.
+     */
+    private static void paintersInit() {
+        _gridPainter.setStyle(Paint.Style.STROKE);
+        _gridPainter.setStrokeWidth(_linesWidth);
+        _gridPainter.setColor(Color.BLACK);
+        _cellsPainter.setStyle(Paint.Style.FILL);
+        _numbersPainter.setTypeface(Typeface.create(Typeface.MONOSPACE, Typeface.NORMAL));
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/nonogramDrawable/NonogramDrawer.java]: The length of this line is over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultImageView.java [hunk 6]
```diff
@@ -164,117 +146,91 @@ public void onGlobalLayout() {
...
-        _centerRelativeX = (nonoCenterX - viewCenterX) / nonoWidth;
-        _centerRelativeY = (nonoCenterY - viewCenterY) / nonoHeight;
+        if (centerDistanceX > drawableWidth / 2f)
+            offsetX += centerDistanceX - drawableWidth / 2f;
+        if (centerDistanceX < -drawableWidth / 2f)
+            offsetX += centerDistanceX + drawableWidth / 2f;
+        if (centerDistanceY > drawableHeight / 2f)
+            offsetY += centerDistanceY - drawableHeight / 2f;
+        if (centerDistanceY < -drawableHeight / 2f)
+            offsetY += centerDistanceY + drawableHeight / 2f;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultImageView.java:225]: Check coding standard for lines 215-222

**Review by maik-nack — APPROVED** [review]
WONDERFUL

**Review by MaximSmolskiy — APPROVED** [review]
Looks great!

**Pull request merged**